src/codeu/chat/client/ClientMessage.java

```diff
@@ -112,6 +112,36 @@ public void addMessage(Uuid author, Uuid conversation, String body) {
     updateMessages(false);
   }
 
+  // Delete message, (m-del command?).TODO
+  public void deleteMessage() {
+    if(current != null) {
+      deleteMessage(current);
+    } else {
+    }
+  }
+
+  public void deleteMessage(Message msg) {
+    if(conversationContents.contains(msg)) {
```

Comment:
Both contains and remove will take linear time, consider replacing with a hash map implementation, be aware that HashMaps are unordered so might need to use an ordered hash map.

See: https://github.com/google/guava for advanced collection libraries.